FileIO/FEFLOWInterface.cpp

```diff
@@ -460,10 +523,15 @@ void FEFLOWInterface::readSuperMesh(std::ifstream &in, const FEM_CLASS &fem_clas
 	}
 }
 
-void FEFLOWInterface::setMaterialID(const FEM_CLASS &fem_class, const FEM_DIM &fem_dim, const std::vector<GeoLib::Polyline*>* lines, std::vector<MeshLib::Element*> &vec_elements)
+void FEFLOWInterface::setMaterialID(const FEM_CLASS &fem_class, const FEM_DIM &fem_dim, const std::vector<GeoLib::Polyline*>* lines, const std::vector<std::vector<std::size_t>> &vec_elementsets, std::vector<MeshLib::Element*> &vec_elements)
 {
-	if (lines && lines->size() > 1)
-	{
+	if (!vec_elementsets.empty()) {
+		for (std::size_t matid =0; matid<vec_elementsets.size(); matid++) {
+			auto &eids = vec_elementsets[matid];
+			for (auto eid : eids)
+				vec_elements[eid-1]->setValue(matid); // Element IDs given by FEFLOW starts from one!
```

Comment:
FYI: setValue takes an unsigned int, losing integer precision implicitly.
Should be handled by changing setValue's parameter to a std::size_t.